### PR TITLE
Store activation status on repo objects, step 2

### PR DESCRIPTION
**Pull request by nworden**
Sequel to PR #627 (motivation explained there). This writes to the new fields when repos are created or edited.

### CODE REVIEW

**Comment by nworden** [comment]
I'll send this to someone in NYC.

File: app/admin.py [hunk 1]
```diff
@@ -225,3 +229,18 @@ def __view_config_to_model_config(self, view_config):
             else:
                 model_config[name] = value
         return model_config
+
+    def __update_repo_status(self, view_config):
+        repo_entity = model.Repo.get_by_key_name(self.env.repo)
+        launch_status = view_config['launch_status']
+        if launch_status == 'staging':
```
Comment (christopherestif) [inline app/admin.py]: just by habbit, I'll ask if it's worth putting these hard-coded strings in some constants somewhere?

File: tests/views/test_admin_create_repo.py [hunk 2]
```diff
@@ -54,13 +54,16 @@ def test_create_repo(self):
...
+        self.assertEqual(
+            repo.activation_status, model.Repo.ActivationStatus.STAGING)
+        self.assertFalse(repo.test_mode)
         # Check a couple of the config fields that are set by default.
         repo_conf = config.Configuration('idaho')
-        assert repo_conf.language_menu_options == ['en', 'fr']
-        assert not repo_conf.launched
-        assert repo_conf.time_zone_abbreviation == 'UTC'
+        self.assertEqual(repo_conf.language_menu_options, ['en', 'fr'])
+        self.assertFalse(repo_conf.launched)
```
Comment (christopherestif) [inline tests/views/test_admin_create_repo.py]: supernit, self.assertIs(..., False) is more specific than self.assertFalse(...):
https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertFalse
I don't know which one is more relevant to your case though

File: app/admin.py (hunk 1 above)
Comment (nworden) [reply app/admin.py]: Done.

File: tests/views/test_admin_create_repo.py (hunk 2 above)
Comment (nworden) [reply tests/views/test_admin_create_repo.py]: Thanks, changed to assertIs.

**Pull request merged**